doc/source/whatsnew/v2.0.0.rst

```diff
@@ -341,6 +341,38 @@ Optional libraries below the lowest tested version may still work, but are not c
 
 See :ref:`install.dependencies` and :ref:`install.optional_dependencies` for more.
 
+Datetimes are now parsed with a consistent format
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In the past, :func:`to_datetime` guessed the format for each element independently. This was appropriate for some cases where a column had a mixed date format - however, it would regularly cause problems for columns where users expected a consistent format but the function would switch formats row-wise. As of version 2.0.0, this behavior is consistent column-wise, and the format is determined by the first non-NA value in the column (unless the user specifies a format, in which case that is used).
```

Comment:
Nit: The use of `row-wise` and `column-wise` makes it sounds like we're working with a DataFrame (which I think isn't the case). Maybe `row-wise` -> `between elements` and `column-wise` -> `across the array of elements`?